GAE/test/org/akvo/flow/xml/FlowXmlObjectReaderTests.java

```diff
@@ -104,70 +104,70 @@
         assertEquals("1.0", testFormDto.getVersion());
-        assertNotEquals(null, testFormDto.getQuestionGroupList());
+        assertNotNull(testFormDto.getQuestionGroupList());
         assertEquals(1, testFormDto.getQuestionGroupList().size());
 
         QuestionGroupDto qg = testFormDto.getQuestionGroupList().get(0);
         assertEquals(1, qg.getOrder());
         assertEquals("The Only Group", qg.getName());
-        assertNotEquals(null, qg.getQuestionMap());
-        assertEquals(1, qg.getQuestionMap().size());
+        assertNotNull(qg.getQuestionList());
+        assertEquals(1, qg.getQuestionList().size());
 
-        QuestionDto q = qg.getQuestionMap().get(1);
-        assertNotEquals(null, q);
+        QuestionDto q = qg.getQuestionList().get(0);
+        assertNotNull(q);
         assertEquals("The Only Question", q.getText());
         assertEquals(QuestionType.FREE_TEXT, q.getType());
         assertTrue(q.getMandatoryFlag());
         assertFalse(q.getLocaleNameFlag());
         assertEquals(24680, q.getKeyId());
-        assertNotEquals(null, q.getTranslationMap());
-        assertEquals(null, q.getTranslationMap().get("fr")); //Should NOT be a French translation
-        assertNotEquals(null, q.getTranslationMap().get("sv")); //Should be a Swedish translation
+        assertNotNull(q.getTranslationMap());
+        assertNull(q.getTranslationMap().get("fr")); //Should NOT be a French translation
+        assertNotNull(q.getTranslationMap().get("sv")); //Should be a Swedish translation
         assertEquals("Den enda frågan", q.getTranslationMap().get("sv").getText());
     }
 
     @Test
     void testParseGrouplessForm() throws IOException {
 
         SurveyDto testFormDto = PublishedForm.parse(GROUPLESS_XML_FORM, true).toDto(); //be strict
-        assertNotEquals(null, testFormDto);
+        assertNotNull(testFormDto);
 
         assertEquals("Foo", testFormDto.getName());
         assertEquals("1.0", testFormDto.getVersion());
-        assertEquals(null, testFormDto.getQuestionGroupList());
+        assertNull(testFormDto.getQuestionGroupList());
     }
 
     @Test
     void testParseQuestionlessForm() throws IOException {
 
         SurveyDto testFormDto = PublishedForm.parse(QUESTIONLESS_XML_FORM, true).toDto(); //be strict
-        assertNotEquals(null, testFormDto);
+        assertNotNull(testFormDto);
 
         assertEquals("Foo", testFormDto.getName());
         assertEquals("1.0", testFormDto.getVersion());
-        assertNotEquals(null, testFormDto.getQuestionGroupList());
+        assertNotNull(testFormDto.getQuestionGroupList());
         assertEquals(1, testFormDto.getQuestionGroupList().size());
 
         QuestionGroupDto qg = testFormDto.getQuestionGroupList().get(0);
         assertEquals(1, qg.getOrder());
         assertEquals("The Empty Group", qg.getName());
-        assertEquals(null, qg.getQuestionMap());
+        assertNull(qg.getQuestionList());
```

Comment:
This made me wonder, in the app, is it ok if you get a `questionList` intead of a `questionMap`?  Will everything still work as expected?

Reply:
The app gets the xml file, but in any case for json we also would need the list. The map is only used to have ordered questions as far as I can tell.